### PR TITLE
restore,streamingccl/streamingest: revert to when tenant was restored

### PR DESCRIPTION
**Pull request by dt**
Previously we started recording the source cluster ID and timestamp when
we restored a backup of a tenant, allowing resuming replication from the
restored backup without needing a full backfill scan. To do this we need
to revert to the timestamp at which the clusters diverged, i.e. when the
source was backed up. However, if that backup is restored at a later
time, reverting to the time at which it was restored has the effect of
reverting to the time that was backed up, since what was restored was
what was backed up. Reverting to the time backed up might be impossible
for the restored cluster, as it might not have even existed at the time
the backup was captured, but reverting to the time at which it was
restored avoid this problem.

Release note: none.
Epic: none.

### CODE REVIEW

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 1]
```diff
@@ -75,10 +75,13 @@ func startDistIngestion(
 
 		log.Infof(ctx, "reverting tenant %s to time %s before starting replication", details.DestinationTenantID, replicatedTime)
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go:76]: Let's move this log message down and log both times

**Review by stevendanna — APPROVED** [review]
A minor logging nit but this otherwise looks pretty good.

**Comment by dt** [comment]
TFTR!

bors r+

**Comment by msbutler** [comment]
For the dull, here's the scenario this patch aims to solve: we have the ability to kickstart a replication stream from a backed up tenant. I.e.:
- t2: Take a backup of Tenant foo
- t3: Write some updates to tenant foo
- t4: Create Tenant baz via restoring tenant foo using the backup taken at t2
- t5: write some updates on baz
- t6: Create a replication stream from foo to baz without an initial scan from t0.

To create a replication stream from a consistent state, we must revert baz to its state at t4, i.e. before received writes at t5. Further, the stream must begin catchup scans from t2, to capture the updates at t3.

**Comment by msbutler** [comment]
@dt it's not clear to me that replication stream that began at t6 will capture the source update at t3, which I think it needs to. My read of `alterTenantRestartReplication` (specifically `pickReplicationResume`) is that the we'll resume the replication stream from t4 (via `resumeTS = from.CutoverTimestamp`), but don't we need to replicate the source update at t3?

And to back up, given that restores of tenants are non-mvcc, why is the patch even necessary?

**Comment by dt** [comment]
blathers backport 24.1

**Pull request merged**